src/gulp/spec.js

```diff
@@ -1,282 +1,31 @@
-    // an array of paths to ignore from coverage reports
-    var ignoreCoverage = opts.ignoreCoverage || [ '**/node_modules/**', '**/__spec__.js' ];
-    // an array to specify what kind of reporters should karma generate
-    var reporters = opts.reporters || ['progress'];
-    // an array to specify what kind of reporters type should karma generate
-    var coverageReporters = opts.coverageReporters || [{ type: 'text-summary' }, { type: 'html' }];
-    // which preprocessors the js files should run through
-    var preProcessors = opts.preProcessors || [ 'browserify' ];
-    // which preprocessors the spec files should run through
-    var specpreProcessors = opts.specpreProcessors || [ 'browserify' ];
-    // where to find the karma config file
-    var configFile = opts.configFile || __dirname + '/karma.conf.js';
-    // defaults the coverage thresholds
-    var coverageThreshold = Object.assign({}, {
-      statements: 100,
-      branches: 100,
-      functions: 100,
-      lines: 100
-    }, opts.coverage);
-    // eslintThreshold for failing build
-    var eslintThreshold = opts.eslintThreshold || null;
-    // Stop build if above eslintThreshold
-    var stopAboveEslintThreshold = !!eslintThreshold;
-    // coverage thresholds for each file
-    var coverageThresholdEachFile = opts.coverageEachFile || {};
-    // if using typescript
-    var typescript = opts.typescript || false;
-    // where the gulp task was ran from
-    var originPath = process.cwd();
-    // array of modules to apply babel transforms to
-    var babelTransforms = opts.babelTransforms || [];
-
-    // check if eslintrc file exists, if not then prompt dev to create one
-    fs.stat(originPath + '/.eslintrc', function(err, stat) {
-      if (err == null) { return }
-      if (err.code == 'ENOENT') {
-        gutil.log(gutil.colors.red("Cannot find '.eslintrc' file"));
-        gutil.log(gutil.colors.white("Create an '.eslintrc' file in the root of your project and add the following code:\n{\n  \"extends\": \"./node_modules/carbon-factory/.eslintrc\"\n}"));
-        process.exit();
-      }
-    });
-
-    if (!argv['skip-eslint']) {
-      preProcessors.unshift('eslint');
-    }
-
-    // prefix the paths with where the gulp task was ran from so the files can
-    // be found from the correct location
-    var src = originPath + path,
-        specSrc = originPath + specs,
-        babelOptions = {
-          babelrc: false, // do not use babelrc files in gulp task
-          extends: originPath + '/node_modules/carbon-factory/.babelrc', // manually set babelrc for gulp task
-          env: {
-            test: {
-              auxiliaryCommentBefore: "istanbul ignore next",
-              plugins: [ [ "istanbul", { exclude: ignoreCoverage } ] ]
-            }
-          }
-        };
-
-    if (babelTransforms.length) {
-      var only = "^((?!node_modules).";
-
-      babelTransforms.forEach((module) => {
-        only += "|(node_modules\/" + module + ")";
```

Comment:
need to ensure this still works in the new world (applying babel transforms to some node_modules)